thunder/benchmarks/benchmark_litgpt.py

```diff
@@ -578,9 +586,33 @@ def benchmark_main(return_metrics_as_json=False, json_path="", **kwargs) -> None
     Runs a training benchmark for lit-GPT models and
     prints out the performance metrics.
     """
-
     benchmark = Benchmark_litGPT(**kwargs)
-    benchmark.train()
+
+    attention_ctx = nullcontext()
+    if sdpa_available and benchmark.use_sdpa:
+        try:
+            attention_ctx = sdpa_kernel(
+                [
+                    SDPBackend.CUDNN_ATTENTION,
+                    SDPBackend.FLASH_ATTENTION,
+                    SDPBackend.EFFICIENT_ATTENTION,
+                    SDPBackend.MATH,
+                ],
+                set_priority=True,  # might fail on older PyTorch versions
+            )
+        except TypeError:
+            # If set_priority isn't supported, fallback
+            attention_ctx = sdpa_kernel(
+                [
+                    SDPBackend.CUDNN_ATTENTION,
+                    SDPBackend.FLASH_ATTENTION,
+                    SDPBackend.EFFICIENT_ATTENTION,
+                    SDPBackend.MATH,
+                ]
+            )
```

Comment:
Looks good but wait maybe better to keep it slightly cleaner:
```suggestion
        backends = [
                SDPBackend.CUDNN_ATTENTION,
                SDPBackend.FLASH_ATTENTION,
                SDPBackend.EFFICIENT_ATTENTION,
                SDPBackend.MATH,
            ]
        kwargs = {}
        if LooseVersion(torch.__version__) >= LooseVersion("2.6.0"):
            kwargs['set_priority'] = True
    
        attention_ctx = sdpa_kernel(backends, **kwargs)
```

and on top `from looseversion import LooseVersion`